invisible_cities/cities/detsim_test.py

```diff
@@ -0,0 +1,277 @@
+        filters = h5out.root.Filters.active_hits.read()
+        np.testing.assert_array_equal(filters["passed"], [False, True])
+
+
+def test_detsim_empty_input_file(ICDATADIR, output_tmpdir):
+
+    PATH_IN  = os.path.join(ICDATADIR    , "nexus_new_kr83m_fast.newformat.sim.h5")
+    PATH_OUT = os.path.join(output_tmpdir, "detsim_test.h5")
+    conf = configure('detsim $ICTDIR/invisible_cities/config/detsim.conf'.split())
+
+    # remove test file rows
+    empty_testfile = os.path.join(output_tmpdir, "empty_testfile.h5")
+    shutil.copyfile(PATH_IN, empty_testfile)
+    with tb.open_file(empty_testfile, "a") as h5file:
+        h5file.root.MC.configuration.remove_rows(0)
+        h5file.root.MC.hits         .remove_rows(0)
+        h5file.root.MC.particles    .remove_rows(0)
+        h5file.flush()
+
+    conf.update(dict(files_in    = empty_testfile,
+                     file_out    = PATH_OUT,
+                     run_number  = 0))
+    result = detsim(**conf)
+
+    assert result.events_in   == 0
+    assert result.evtnum_list == []
+
+
+def test_detsim_exact(ICDATADIR, output_tmpdir):
+
+    PATH_IN   = os.path.join(ICDATADIR    , "nexus_new_kr83m_fast.newformat.sim.h5")
+    PATH_OUT  = os.path.join(output_tmpdir, "detsim_test.h5")
+    PATH_TRUE = os.path.join(ICDATADIR    , "detsim_new_kr83m_fast.newformat.sim.h5")
+    conf = configure('detsim $ICTDIR/invisible_cities/config/detsim.conf'.split())
+
+    buffer_params = dict()
+    buffer_params["length"]      = 800 * units.mus
+    buffer_params["pmt_width"]   = 100 * units.ns
+    buffer_params["sipm_width"]  =   1 * units.mus
+    buffer_params["max_time"]    =  10 * units.ms
+    buffer_params["pre_trigger"] = 100 * units.mus
+    buffer_params["trigger_thr"] = 0
+
+    conf.update(dict(files_in      = PATH_IN,
+                     file_out      = PATH_OUT,
+                     run_number    = 0,
+                     event_range   = all_events,
+                     buffer_params = buffer_params))
+    np.random.seed(1234)
+    result = detsim(**conf)
+
+    tables = ["pmtrd", "sipmrd",
+              "Run/eventMap", "Run/events", "Run/runInfo",
+              "MC/configuration", "MC/event_mapping", "MC/hits",
+              "MC/particles", "MC/sns_positions", "MC/sns_response",
+              "Filters/active_hits"]
+
+    with tb.open_file(PATH_TRUE) as true_output_file:
+        with tb.open_file(PATH_OUT) as output_file:
+            for table in tables:
+                assert hasattr(output_file.root, table)
+                got      = getattr(     output_file.root, table)
+                expected = getattr(true_output_file.root, table)
+                assert_tables_equality(got, expected)
+
+
+def test_detsim_exact_time_translation(ICDATADIR, output_tmpdir):
+
+    PATH_IN   = os.path.join(ICDATADIR    , "nexus_new_kr83m_fast.newformat.sim.h5")
+    PATH_OUT  = os.path.join(output_tmpdir, "detsim_test.h5")
+
+    physics_params = dict()
```

Comment:
Again, copy already existing dictionary inside `conf` and only modify needed values